public/src/managers/ServerProvider.ts

```diff
@@ -81,114 +81,116 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
         }
-        return result.data.getIsauthorized();
+        return false;
+
     }
 
     public async createNewCourse(course: Course): Promise<Course | Status> {
         const result = await this.grpcHelper.createCourse(course);
-        if (result.status.getCode() !== 0 || !result.data) {
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return result.status;
         }
         return result.data;
     }
 
-    public async getCourse(ID: number): Promise<Course | null> {
-        const result = await this.grpcHelper.getCourse(ID);
-        if (result.status.getCode() !== 0 || !result.data) {
+    public async getCourse(courseID: number): Promise<Course | null> {
+        const result = await this.grpcHelper.getCourse(courseID);
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return null;
         }
         return result.data;
     }
 
     public async updateCourse(course: Course): Promise<Void | Status> {
         const result = await this.grpcHelper.updateCourse(course);
-        if (result.status.getCode() !== 0 || !result.data) {
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return result.status;
         }
         return new Void();
     }
 
-    public async createGroup(courseID: number, name: string, users: number[]): Promise<Group | Status> {
-        const result = await this.grpcHelper.createGroup(courseID, name, users);
-        if (result.status.getCode() !== 0 || !result.data) {
+    public async createGroup(courseID: number, groupName: string, users: number[]): Promise<Group | Status> {
+        const result = await this.grpcHelper.createGroup(courseID, groupName, users);
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return result.status;
         }
         return result.data;
     }
 
     public async getCourseGroups(courseID: number): Promise<Group[]> {
         const result = await this.grpcHelper.getGroups(courseID);
-        if (result.status.getCode() !== 0 || !result.data) {
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return [];
         }
         return result.data.getGroupsList();
     }
 
     public async getGroupByUserAndCourse(courseID: number, userID: number): Promise<Group | null> {
         const result = await this.grpcHelper.getGroupByUserAndCourse(courseID, userID);
-        if (result.status.getCode() !== 0 || !result.data) {
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return null;
         }
         return result.data;
     }
 
-    public async updateGroupStatus(groupID: number, states: Group.GroupStatus): Promise<Status> {
-        const result = await this.grpcHelper.updateGroupStatus(groupID, states);
+    public async updateGroupStatus(groupID: number, status: Group.GroupStatus): Promise<Status> {
+        const result = await this.grpcHelper.updateGroupStatus(groupID, status);
```

Comment:
Shouldn't this also check the status code?

Reply:
We don't want to filter out failed requests here, they carry error messages to show to the user.

Reply:
okay, maybe add a comment about that.